x/concentrated-liquidity/swaps.go

```diff
@@ -591,6 +602,13 @@ func (k Keeper) computeInAmtGivenOut(
 			if err != nil {
 				return sdk.Coin{}, sdk.Coin{}, 0, sdk.Dec{}, sdk.Dec{}, sdk.Dec{}, err
 			}
+
+			// Move next tick iterator to the next tick as the tick is crossed.
+			if !nextTickIter.Valid() {
+				return sdk.Coin{}, sdk.Coin{}, 0, sdk.Dec{}, sdk.Dec{}, sdk.Dec{}, types.RanOutOfTicksForPoolError{PoolId: poolId}
+			}
```

Comment:
I don't think we need to check this again here right?

Reply:
I think you're correct. Updated: [74bd8ac](https://github.com/osmosis-labs/osmosis/pull/5248/commits/74bd8acce14f793d5984d1d1bf1e4e58959297d0)